src/App.tsx

```diff
@@ -21,14 +13,41 @@ const styles = StyleSheet.create({
   },
 });
 
-interface IProps {}
+interface IProps {
+  decreaseCount: (counterId: number) => void,
+  increaseCount: (counterId: number) => void,
+  counters: ICounter[]
+}
 interface IState {}
-export default class App extends Component<IProps, IState> {
+
+class App extends Component<IProps, IState> {
   public render() {
     return (
       <View style={styles.container}>
-        <Counter />
+      {this.renderCounters()}
       </View>
     );
   }
+
+  private renderCounters() {
+    return this.props.counters.map((counter: ICounter, counterId: number) => {
+      return <Counter key={counterId}
+                      count={counter.count}
+                      decreaseCount={() => this.props.decreaseCount(counterId) }
+                      increaseCount={() => this.props.increaseCount(counterId) } />;
+    });
+  }
 }
+
+const mapDispatchToProps = (dispatch: any) => {
+  return {
+    decreaseCount: (counterId: number) => dispatch({ type: 'DECREASE_COUNT', counterId }),
```

Comment:
Use the redux function `bindActionCreators` here to remove a bunch of this boilerplate.